src/v/pandaproxy/schema_registry/json.cc

```diff
@@ -630,6 +630,47 @@ bool is_object_superset(
       pj{newer})));
 }
 
+bool is_enum_superset(json::Value const& older, json::Value const& newer) {
+    auto older_it = older.FindMember("enum");
+    auto newer_it = newer.FindMember("enum");
+    auto older_is_enum = older_it != older.MemberEnd();
+    auto newer_is_enum = newer_it != newer.MemberEnd();
+
+    if (!older_is_enum && !newer_is_enum) {
+        // both are not an "enum" schema, compatible
```

Comment:
nitpick: I guess this actually means "not necessarily incompatible"?

Reply:
right, "compatible for what `enum` is concerned". 
It's legal to define this schema
```
reader = { "type": "integer", "enum": ["a string"] }
writer = { "type": "boolean", "enum": ["a string"] }
```
but  although these two validate the same data (none), they are not compatible according to these classification rules